delta-lake/src/main/java/org/apache/iceberg/delta/SupportSnapshotFromDeltaLake.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.delta;
+
+/**
+ * An API that should be implemented by query engine integrations that want to support snapshotting
+ * from Delta Lake table to Iceberg table.
+ */
+public interface SupportSnapshotFromDeltaLake {
+
+  /** Initiates an action to migrate an existing Delta Lake table to Iceberg. */
+  default SnapshotDeltaLakeTable migrateDeltaLakeTable(String tableIdent, String deltaS3Location) {
```

Comment:
1. as said in the previous comment, this should probably be `snapshotDeltaLakeTable`.
2. for the arguments of the method, we can consult the `ActionsProvider` interface, I think we only need 1 input which is `String sourceTableLocation`
3. we can offer a default implementation for the interface, which is to return the Base class you implemented

Reply:
Thank you for your suggestions. I added `as`, `icebergCatalog` and `deltaLakeConfiguration` to the API and also added Precondition checks in `execute()` to make the three property required. 

One thing I noticed that in `SnapshotTableSparkAction`, it does not allow the iceberg table to locate in the same place as the source table:
https://github.com/apache/iceberg/blob/ceaddbfa33fce56f091a8b656e11c9a3cab9b5d0/spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/SnapshotTableSparkAction.java#L206-L213

Do you think we need to do the same here? I think it may be fine in snapshoting delta lake case because all metdata of delta lake table resides in `.../<table name>/_delta_log` while those of iceberg table resides in `.../<table name>/metadata`